### PR TITLE
Elaborate documentation for notification and Illumina runscanner

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java [hunk 1]
```diff
@@ -15,6 +15,16 @@
...
 
+/**
+ * A "run" as seen by Run Scanner
+ *
+ * Run Scanner collects information about runs and reports them to MISO for incorporation. This class is distinct from {@link RunDto}
+ * because there is not a direct mapping between runs in MISO and runs as they can be detected on disk ("notifications"). First, runs have
+ * added information such as change logs, security profiles, and notes that notifications should never include. Second, the notification is
+ * actually a composite of run, container, and partition information and this has to be expanded when written to MISO in a non-trivial way.
+ * Finally, notifications contain best-effort information gleaned from the sequencer output, but it is not always possible to correctly
+ * detect information; if a human changes certain properties of a run, it should be updated from automation.
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java]: `should be updated from automation` -> `should not be overwritten by automation`?

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java [hunk 2]
```diff
@@ -63,6 +97,15 @@ public void setLaneCount(int laneCount) {
     this.laneCount = laneCount;
   }
 
+  /**
+   * Get the status of the run.
+   * 
+   * This the current status of the run. This may be impossible to correctly detect depending on the platform. MISO will not record the
+   * completion date if the health is not done. If the status has been changed by a human, any updates provided in notifications are
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java]: `if the health is not [enumeration of completed states]` or `if the health is "Running", "Unknown", or "Started"` may be more clear (or move the `@see HealthType#isDone()` to a parenthetical at the end of that sentence).

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java [hunk 3]
```diff
@@ -71,6 +114,12 @@ public void setHealthType(HealthType healthType) {
     this.healthType = healthType;
   }
 
+  /**
+   * Get the file path to the sequencing output
+   * 
+   * MISO only present this data to the user and possibly to a downstream analysis pipeline. It will never attempt to read from this
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java]: `presents`

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java [hunk 4]
```diff
@@ -95,6 +152,9 @@ public void setSoftware(String software) {
     this.software = software;
   }
 
+  /**
+   * Gets the time when the sequencer run started, if known.
```
Comment (callunity) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/NotificationDto.java]: `Get`

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [IlluminaNotificationDto.java#L23](https://github.com/TGAC/miso-lims/blob/4bd68ca1e78e196db563a5c0f3babcbb3f75bd4a/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java#L23): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [EditRunController.java#L185](https://github.com/TGAC/miso-lims/blob/4bd68ca1e78e196db563a5c0f3babcbb3f75bd4a/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L185): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [EditRunController.java#L194](https://github.com/TGAC/miso-lims/blob/4bd68ca1e78e196db563a5c0f3babcbb3f75bd4a/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L194): Define a constant instead of duplicating this literal "partitionNames" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [EditRunController.java#L199](https://github.com/TGAC/miso-lims/blob/4bd68ca1e78e196db563a5c0f3babcbb3f75bd4a/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L199): Refactor the code so this stream pipeline is used. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by apmasell** [comment]
Updated.

File: docs/_posts/2016-01-11-admin-manual.md [hunk 5]
```diff
@@ -312,34 +312,66 @@ Java package. To create a new naming scheme option, create a new class in this p
 Extending the functionality to validate and/or generate additional fields is possible, but will
 require modifications at the Service layer as well.
 
-# Setting Up the Notification Server
-The notification server is a Java daemon that scans the paths containing
+# Setting Up the Run Scanner
+The notification server is a webservice  that scans the paths containing
```
Comment (djcooke) [inline docs/_posts/2016-01-11-admin-manual.md]: run scanner

File: docs/_posts/2016-01-11-admin-manual.md [hunk 6]
```diff
@@ -312,34 +312,66 @@ Java package. To create a new naming scheme option, create a new class in this p
...
+
+Create a file called `ROOT.xml` in the following directory
+`$CATALINA_HOME/conf/Catalina/localhost`, creating the directory if necessary,
+and populate it with the following information:
 
-Create the default configuration:
 
     mkdir /srv/notification-server
     cp $MISO_SRC/notification-server/service/notification.properties /srv/notification-server
     cp $MISO_SRC/notification-server/service/miso-notification.service /etc/systemd/system
```
Comment (djcooke) [inline docs/_posts/2016-01-11-admin-manual.md]: remove

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [IlluminaNotificationDto.java#L23](https://github.com/TGAC/miso-lims/blob/fa5c525e61cf55523d521bea64557032b2b59b4e/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java#L23): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [EditRunController.java#L185](https://github.com/TGAC/miso-lims/blob/fa5c525e61cf55523d521bea64557032b2b59b4e/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L185): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [EditRunController.java#L194](https://github.com/TGAC/miso-lims/blob/fa5c525e61cf55523d521bea64557032b2b59b4e/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L194): Define a constant instead of duplicating this literal "partitionNames" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [EditRunController.java#L199](https://github.com/TGAC/miso-lims/blob/fa5c525e61cf55523d521bea64557032b2b59b4e/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L199): Refactor the code so this stream pipeline is used. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by apmasell** [comment]
Fixed. The default port is whatever they configure it to. I guess 8080.

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [IlluminaNotificationDto.java#L23](https://github.com/TGAC/miso-lims/blob/3d7012ac808d182c74e5fb5d3cf37d0cf7da0075/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDto.java#L23): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [EditRunController.java#L185](https://github.com/TGAC/miso-lims/blob/3d7012ac808d182c74e5fb5d3cf37d0cf7da0075/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L185): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [EditRunController.java#L194](https://github.com/TGAC/miso-lims/blob/3d7012ac808d182c74e5fb5d3cf37d0cf7da0075/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L194): Define a constant instead of duplicating this literal "partitionNames" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [EditRunController.java#L199](https://github.com/TGAC/miso-lims/blob/3d7012ac808d182c74e5fb5d3cf37d0cf7da0075/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java#L199): Refactor the code so this stream pipeline is used. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3958)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**